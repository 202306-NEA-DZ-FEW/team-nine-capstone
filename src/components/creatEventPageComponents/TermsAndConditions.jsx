import React from "react";

export default function TermsAndConditions() {
    return <div>terms and conidtions</div>;
}
