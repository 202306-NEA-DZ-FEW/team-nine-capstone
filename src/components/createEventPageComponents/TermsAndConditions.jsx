import React from "react";

export default function TermsAndConditions() {
    return (
        <>
            <div>
                <h1>Welcome to the Pebbler Community</h1>

                <h2>1. Compassion and Empathy</h2>
                <p>
                    Approach charity and volunteering with a deep sense of
                    compassion and empathy for those in need.
                </p>

                <h2>2. Voluntary Service</h2>
                <p>
                    Volunteering is an act of free will. It should not be
                    coerced or forced.
                </p>

                <h2>3. Respect and Dignity</h2>
                <p>
                    Treat all individuals with respect and dignity, regardless
                    of their background, race, religion, or social status.
                </p>

                <h2>4. Equality and Inclusivity</h2>
                <p>
                    Ensure that charity efforts and volunteer opportunities are
                    open to everyone. Promote inclusivity and diversity.
                </p>

                <h2>5. Transparency and Accountability</h2>
                <p>
                    Maintain transparency in financial transactions and the
                    allocation of resources. Be accountable for the funds and
                    donations received.
                </p>

                <h2>6. Sustainability</h2>
                <p>
                    Aim for long-term, sustainable solutions rather than
                    short-term fixes. Address the root causes of social issues.
                </p>

                <h2>7. Partnerships and Collaboration</h2>
                <p>
                    Collaborate with other charities, nonprofits, and community
                    organizations for more significant positive impacts.
                </p>

                <h2>8. Effective Resource Allocation</h2>
                <p>
                    Allocate resources wisely to maximize the impact of your
                    efforts.
                </p>

                <h2>9. Needs Assessment</h2>
                <p>
                    Conduct thorough needs assessments to understand the
                    specific requirements of the community or individuals you
                    are serving.
                </p>

                <h2>10. Skill-Based Volunteering</h2>
                <p>
                    Encourage volunteers to use their unique skills and
                    expertise for more targeted and impactful contributions.
                </p>
            </div>
        </>
    );
}
