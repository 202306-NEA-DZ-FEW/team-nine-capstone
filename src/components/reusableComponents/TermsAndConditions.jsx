import { useTranslation } from "next-i18next";

function TermsAndConditions({ closeModal }) {
    const { t } = useTranslation("common");
    return (
        <div className='fixed inset-0 flex items-center justify-center z-50 backdrop-blur-md'>
            <div className='bg-white text-black w-full md:w-1/2 rounded-lg p-4 max-h-[80vh] overflow-y-auto'>
                <div className='text-2xl font-bold mb-4'>{t("t&c.title")}</div>
                <div className='relative text-start'>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t1")}
                    </div>
                    <div className='text-base'>{t("t&c.b1")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t2")}
                    </div>
                    <div className='text-base'>{t("t&c.b2")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t3")}
                    </div>
                    <div className='text-base'>{t("t&c.b3")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t4")}
                    </div>
                    <div className='text-base'>{t("t&c.b4")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t5")}
                    </div>
                    <div className='text-base'>{t("t&c.b5")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t6")}
                    </div>
                    <div className='text-base'>{t("t&c.b6")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t7")}
                    </div>
                    <div className='text-base'>{t("t&c.b7")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t8")}
                    </div>
                    <div className='text-base'>{t("t&c.b8")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t9")}
                    </div>
                    <div className='text-base'>{t("t&c.b9")}</div>
                    <div className='text-xl underline font-semibold'>
                        {t("t&c.t10")}
                    </div>
                    <div className='text-base'>{t("t&c.b10")}</div>
                </div>
                <button
                    onClick={() => closeModal(false)}
                    className='border-transparent bg-amber-500 hover:bg-green-500 focus:bg-green-700 rounded-lg px-4 py-2 mt-4'
                >
                    {t("t&c.button")}
                </button>
            </div>
        </div>
    );
}

export default TermsAndConditions;
