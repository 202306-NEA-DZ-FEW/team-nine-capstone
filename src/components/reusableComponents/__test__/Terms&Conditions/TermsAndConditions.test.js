import renderer from "react-test-renderer";

import TermsAndConditions from "../../TermsAndConditions";

it("renders correctly", () => {
    const tree = renderer.create(<TermsAndConditions />).toJSON();
    expect(tree).toMatchSnapshot();
});
