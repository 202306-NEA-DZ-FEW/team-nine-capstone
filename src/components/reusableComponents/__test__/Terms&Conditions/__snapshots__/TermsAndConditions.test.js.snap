// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed inset-0 flex items-center justify-center z-50 backdrop-blur-md"
>
  <div
    className="bg-white text-black w-full md:w-1/2 rounded-lg p-4 max-h-[80vh] overflow-y-auto"
  >
    <div
      className="text-2xl font-bold mb-4"
    >
      t&c.title
    </div>
    <div
      className="relative text-start"
    >
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t1
      </div>
      <div
        className="text-base"
      >
        t&c.b1
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t2
      </div>
      <div
        className="text-base"
      >
        t&c.b2
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t3
      </div>
      <div
        className="text-base"
      >
        t&c.b3
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t4
      </div>
      <div
        className="text-base"
      >
        t&c.b4
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t5
      </div>
      <div
        className="text-base"
      >
        t&c.b5
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t6
      </div>
      <div
        className="text-base"
      >
        t&c.b6
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t7
      </div>
      <div
        className="text-base"
      >
        t&c.b7
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t8
      </div>
      <div
        className="text-base"
      >
        t&c.b8
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t9
      </div>
      <div
        className="text-base"
      >
        t&c.b9
      </div>
      <div
        className="text-xl underline font-semibold"
      >
        t&c.t10
      </div>
      <div
        className="text-base"
      >
        t&c.b10
      </div>
    </div>
    <button
      className="border-transparent bg-amber-500 hover:bg-green-500 focus:bg-green-700 rounded-lg px-4 py-2 mt-4"
      onClick={[Function]}
    >
      t&c.button
    </button>
  </div>
</div>
`;
